core/unix/pcprofile.c

```diff
@@ -548,10 +548,12 @@ pcprofile_results(thread_pc_info_t *info)
                     print_file(info->file, " dispatch\n");
                 } else if (e->whereami == DR_WHERE_MONITOR) {
                     print_file(info->file, " monitor\n");
-                } else if (e->whereami == DR_WHERE_SIGNAL_HANDLER) {
-                    print_file(info->file, " signal handler\n");
                 } else if (e->whereami == DR_WHERE_SYSCALL_HANDLER) {
                     print_file(info->file, " syscall handler\n");
+                } else if (e->whereami == DR_WHERE_SIGNAL_HANDLER) {
+                    print_file(info->file, " signal handler\n");
```

Comment:
I see that the order between `DR_WHERE_SYSCALL_HANDLER` and `DR_WHERE_SIGNAL_HANDLER` has changed too. Is that intended?

Reply:
Yes, I made it match the enum order, which makes it easier to see whether anything is missing.

Reply:
If this were the only dispatch I would use a case and the compiler could tell us if we missed something, but two of the cases are large and handled separately so I abandoned that plan.  Plus I think that only works if we have no default case.